feat(breaking): 10.0 Release

JessicaSachs opened the pull request:
BREAKING CHANGE: 10.0 release

# ⚠️ DO NOT SQUASH AND MERGE. USE A MERGE COMMIT

*Targeted release date: June 1st, 2022*

chrisbreiding commented on packages/launchpad/package.json:
```diff
@@ -0,0 +1,111 @@
+    "windicss-analysis": "^0.3.4",
+    "wonka": "^4.0.15"
+  },
+  "files": [
```
This should only need to include `dist`. Otherwise, source files will be unnecessarily included in the binary.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/cypress-io/cypress?pullRequest=18924) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/cypress-io/cypress?pullRequest=18924) before we can accept your contribution.<br/>**12** out of **13** committers have signed the CLA.<br/><br/>:white_check_mark: JessicaSachs<br/>:white_check_mark: tgriesser<br/>:white_check_mark: lmiller1990<br/>:white_check_mark: estrada9166<br/>:white_check_mark: ImCesar<br/>:white_check_mark: ZachJW34<br/>:white_check_mark: elevatebart<br/>:white_check_mark: mapsandapps<br/>:white_check_mark: BlueWinds<br/>:white_check_mark: flotwig<br/>:white_check_mark: marktnoonan<br/>:white_check_mark: chrisbreiding<br/>:x: Mark Noonan<br/><hr/>**Mark Noonan** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/cypress-io/cypress?pullRequest=18924) it.</sub>

emilyrohrbough commented on packages/runner-ct/static/index.html:
```diff
@@ -7,20 +7,16 @@
       // set a global so we know the 'top' window
       window.__Cypress__ = true
-
-      setTimeout(function(){
```
This was pretty much 1:1 with `packages/runner/static/index.html`. Did we need to keep this timeout or do we need to remove the timeout in `packages/runner/static/index.html`.?

ZachJW34 committed:
Merge pull request #21131 from cypress-io/zachw/merge-develop-10.0-4-19

davidmunechika committed:
fix(unify): Remove 'Reconfigure' dropdown from Testing Type chooser (#21063)

Co-authored-by: Ryan Manuel <ryanm@cypress.io>
Co-authored-by: Zachary Williams <ZachJW34@gmail.com>

estrada9166 committed:
fix: do not highlight preExtension if selected option is renameFolder (#21121)

* fix: do not highlight preExtension if selected option is renameFolder

* Fix ts

* Do not change ext

ZachJW34 committed:
chore: add dev-servers as deps to server to be included in the binary (#21142)

Co-authored-by: Tim Griesser <tgriesser10@gmail.com>

ZachJW34 committed:
feat: add next preset to webpack-dev-server-fresh (#21069)

Co-authored-by: Tim Griesser <tgriesser10@gmail.com>

marktnoonan committed:
fix: label text should be clickable to toggle snapshot highlight (#21122)

* update component

* update test

* update component test

* remove unused label text

* add cursor:pointer styles to label

Commit:
chore: Move component-index generation to scaffold-config package (#21090)

* chore: Move component-index generation to scaffold-config package

* Review fixes

* Fix review comments for realzies this time

JessicaSachs committed:
chore: cutting over system-tests and Cypress to use the new CT Object API (#21079)

* removing vite-dev-server local dependency from react-vite-ts-configured system test

* moving some CRA examples over to use the object api for setup

* fixing issue where function API was broken by object API for cy config + devservers

* adding deeply nested react import to project-fixtures for cra

* finishes cutting over cypress/react for sys tests

* chore: adding circle for this feature branch

* chore: moving over many vue + vite system tests to use object API instead of function API (#21080)

* doing webpack-dev-server cutovers

* removing more webpack-dev-server refrences

* fixing snapshots

* bumping yarn.lock

* wip

* fix test

* fix assertion

Co-authored-by: Lachlan Miller <lachlan.miller.1990@outlook.com>

* feat: removing all references for "fresh" dev servers (webpack-dev-server-fresh and vite-dev-server-fresh) (#21094)

Co-authored-by: Lachlan Miller <lachlan.miller.1990@outlook.com>
Co-authored-by: Zachary Williams <ZachJW34@gmail.com>

* chore: add dev-servers as deps to server to be included in the binary (#21091)

* fix bad merge

* fix next types and webpack-dev-server- resolve

Co-authored-by: Lachlan Miller <lachlan.miller.1990@outlook.com>
Co-authored-by: Zachary Williams <ZachJW34@gmail.com>

lmiller1990 committed:
feat: add alpha/beta labels in launchpad (#21111)

* add alpha label

* add ribbon for beta CT

* fix types

* add supportStatus field and better support for a11y

* fix types

* fix tests

tbiethman committed:
Simplifying key usage

tbiethman committed:
Merge branch '10.0-release' of https://github.com/cypress-io/cypress into tbiethman/fix/UNIFY-1504-disable-browsers

tbiethman committed:
Merge pull request #21120 from cypress-io/tbiethman/fix/UNIFY-1504-disable-browsers

fix: Disabling browser dropdown in runner while tests are running

davidmunechika committed:
fix(unify): Specs list within the test runner should be collapsed by default (#21103)

* change default specs list open

* update e2e tests

* revert test change

* change defaults

* update reporter tests

* update e2e test

* remove cy.wait

Co-authored-by: Mark Noonan <mark@cypress.io>

JessicaSachs committed:
fix: stop ResizeObserver error within the SelectorPlayground from throwing error (#21143)

* fixing the resizeobserver error within the SelectorPlayground

* spellcheck off

* unnecessary ref

lmiller1990 committed:
remove old test

Commit:
feat: UI updates to scaffolded files (#21155)

tbiethman committed:
fix: Long Cypress verification times  (#21174)

* Fixing some electron

* More electron refs

* Updating system test snapshots. Updating mocha timeout for module-api tests.

* Updating magic require with comment

* Updating system test snapshot again

* Removing dead code

BlueWinds committed:
chore: Remove storybook related data-context and tests

ZachJW34 committed:
fix: supportFile path and supportFile false for vite on windows (#21156)

Co-authored-by: Tim Griesser <tgriesser10@gmail.com>

BlueWinds committed:
Fix tests from last commit

tgriesser committed:
feat: Auto-merge config when adding new testing type (#21152)

* wip on AST merging Cypress Config

* Adding test coverage & better implementation for config merging

* fix types, use new addToCypressConfig for projectId

* refactor: separate export for browser

* printing w/ recast

* don't remove prettier in this PR

* Update packages/config/src/ast-utils/addToCypressConfig.ts

* make comment more robust

Co-authored-by: Ryan Manuel <ryanm@cypress.io>

lmiller1990 committed:
fix: update dev server error message (#21181)

* fix: update dev server error message

* revert change

* remove obsolete test

* revert change

* update errors

* rename error

Commit:
Merge branch '10.0-release' into UNIFY-1557-remove-storybook-remnants

BlueWinds committed:
Removing leftover .storbook directories

Commit:
Merge pull request #21191 from cypress-io/UNIFY-1557-remove-storybook-remnants

chore: Remove storybook related data-context and tests

ryanthemanuel committed:
fix: retrying config loading errors that lead to new errors freezes initialization (#21176)

* fix: trying again on errors that results in new errors

* fix: trying again on errors that results in new errors

* Refactoring and code clean up for private lifecycle functions

* Refactoring and code clean up for private lifecycle functions

* Fix test flakiness

* More code cleanup and refactoring

* Fix unit tests

* Add unit tests and more refactoring

* Remove comments and unneccessary asyncs

* PR comments

* PR comments

* Update packages/data-context/src/data/ProjectLifecycleManager.ts

tgriesser committed:
fix: dedicated socket.io namespace for graphql-refetch messages (#21207)

* add failing test

* fix: add data-context dedicated socket.io namespace

* better comments & test name to explain what this is testing

ryanthemanuel committed:
chore: percy flake around version upgrade modal (#21210)

AtofStryker committed:
Merge branch 'develop' into md-10.0-merge

emilyrohrbough committed:
feat(Unification): Update Command Group UI (#20465)

estrada9166 committed:
fix: use prettier to format migrated config (#21227)

* fix: use prettier to format migrated config

* Update with feedback

* fix tests

tbiethman committed:
feat: Additional preparations for Chrome's immutable domain updates (#21208)

* feat: Updating client route response headers to continue support for mutable domains

* Updating route integration test

estrada9166 committed:
fix: update not supported plugins errors (#21169)

* fix: update not supported plugins errors

* Add tests

* Add tests

* Fix test

* Update with feedback

emilyrohrbough committed:
feat: update .within to use internal log group api (#20980)

tbiethman committed:
feat: Presenting success banner after dependency installation (#21235)

* Adding banner on dep installation success. Updating copy to match figma.

* Adding component/e2e test coverage

* Flipping polarity on this one

* Update packages/launchpad/cypress/e2e/scaffold-project.cy.ts

Co-authored-by: Rachel <rachel@cypress.io>

* Updates for PR comments

Co-authored-by: Rachel <rachel@cypress.io>

lmiller1990 committed:
fix: subscribe to changes in cypress.config.js in app (#21160)

* chore: work on build

* subscribe to config changes

* add test

* extract required values from legacy config

* add testingType and isComponentTesting to config

* update cypress config

* update runner config on window and re-execute after config changes

* revert change

* revert code

* revert

* revert file

* update stub types

ZachJW34 committed:
fix: automation disconnected error appearing when switching browsers (#21247)

davidmunechika committed:
fix(unify): Sort all resolved config values alphabetically (#21190)

Co-authored-by: Barthélémy Ledoux <bart@cypress.io>

davidmunechika committed:
fix: Open generated spec file preview by default (#21225)

Co-authored-by: Zachary Williams <ZachJW34@gmail.com>
Co-authored-by: Tyler Biethman <tbiethman@users.noreply.github.com>

AtofStryker committed:
Merge branch '10.0-release' of github.com:cypress-io/cypress into md-10.0-merge

AtofStryker committed:
Merge branch '10.0-release' of github.com:cypress-io/cypress into md-10.0-merge

AtofStryker committed:
fix circle config to remove old workflow filters from develop merge

ZachJW34 committed:
chore: add alpha nuxt tag (#21229)

Co-authored-by: Tim Griesser <tgriesser10@gmail.com>

ZachJW34 committed:
fix: restart dev-server on config change (#21212)

* fix: restart dev-server on config change

* close dev server before cp is spawned

* fix test that is failing... not sure why

* cleanup unsued close events

* remove wait

* add back in close for unit tests

Co-authored-by: Tim Griesser <tgriesser10@gmail.com>

ryanthemanuel committed:
fix: update icon for record key in dashboard settings (#21282)

estrada9166 committed:
fix: use correct configFile while creating config (#21273)

AtofStryker committed:
Merge branch 'develop' into md-10.0-merge

estrada9166 committed:
fix: prevent showing project cache if user is not member (#21154)

* Add failing test

* Get latest data on project

* Logout on 401

* remove unused async

* Update error handling

* Fix TS

Co-authored-by: Tim Griesser <tgriesser10@gmail.com>
Co-authored-by: Lachlan Miller <lachlan.miller.1990@outlook.com>

tgriesser committed:
refactor: cleanup types for SpecList component (#21272)

Co-authored-by: Mark Noonan <mark@cypress.io>

AtofStryker committed:
Merge branch 'develop' of github.com:cypress-io/cypress into md-10.0-merge

AtofStryker committed:
Merge branch '10.0-release' of github.com:cypress-io/cypress into md-10.0-merge

AtofStryker committed:
fix driver tests to glob correct specs in e2e directory over integration directory

estrada9166 committed:
fix: Open file in IDE on windows (#21278)

* fix: Open file in IDE on windows

* Use Cypress.config

AtofStryker committed:
fix typings errors introduced by the multi-domain merge

AtofStryker committed:
remove trailing white spaces in cypress.d.ts cli types

AtofStryker committed:
remove test-mocha-snapshot-job

AtofStryker committed:
Merge branch '10.0-release' of github.com:cypress-io/cypress into md-10.0-merge

AtofStryker committed:
ignore new url constructor ts compiler issues as it ships natively with
node and out supported browsers

ryanthemanuel committed:
fix: moving away from runner and disconnecting websockets stops listening for changes to webpack build (#21281)

AtofStryker committed:
fix navigation.cy.js to account for cypress.config.ts in cross origin failure test

AtofStryker committed:
fix chrome CRI client tests by referencing this.pageCriCLient over this.criClient (variable name change)

AtofStryker committed:
update from snakecase verify mocha results to kebab case

ryanthemanuel committed:
chore: fix build due to mis merge (#21301)

* Update cypress-in-cypress-component.cy.ts

* Empty commit to get a new percy nonce

AtofStryker committed:
update web_security e2e spec test with test skipped with a TODO to fix

AtofStryker committed:
fix session e2e snapshot

AtofStryker committed:
update app settings tests to use experimentalSessionAndOrigin over experimentalSessionSupport and update en-US i18n, which seems to take the place of cypress.schema.json

AtofStryker committed:
update system test flags for cy-in-cy tests to use experimentalSessionAndOrigin over experimentalSessionSupport

AtofStryker committed:
update npm vite-dev-server config and frontend-shared cypress fixtures to use experimentalSessionAndOrigin flag

lmiller1990 committed:
fix: various minor migration bugs (#21195)

* correct indentation

* update formatting

* add comments

* update tsconfig.json

* Update packages/data-context/src/sources/migration/codegen.ts

Co-authored-by: Blue F <blue@cypress.io>

* Update packages/data-context/src/sources/migration/codegen.ts

Co-authored-by: Blue F <blue@cypress.io>

Co-authored-by: Blue F <blue@cypress.io>

emilyrohrbough committed:
fix: fix nested command log styles (#21276)

AtofStryker committed:
remove config/lib from config package (not applicable in 10.0) and move session support removed options to config/src

AtofStryker committed:
try to make the web security snapshot happy

AtofStryker committed:
fix unit test invocation to remove typo

AtofStryker committed:
fix cypress binary root (not sure how this one happened...)

AtofStryker committed:
update index snapshot for @packages/config

flotwig committed:
fix(launchpad): avoid crash when authenticating before selecting project (#21234)

marktnoonan committed:
fix: reporter resizing in small viewports and AUT scaling when specs list is open (#21254)

tgriesser committed:
chore: upgrade urql dependencies (#21302)

* chore: upgrade urql dependencies

* fix

AtofStryker committed:
opt for getPRimary over current in getConfig() in the project base so set remote to primary in the driver

marktnoonan committed:
feat: reflect the GQL baseError state in App, not just launchpad (#21157)

Co-authored-by: Tim Griesser <tgriesser10@gmail.com>

estrada9166 committed:
feat: Update migration for projects with projectId (#21193)

* feat: Update migration for projects with projectId

* Add tests

* Update prettier require, add test for custom specPattern

* Revert prettier changes

* Update test

* Add defaultMigrationTransformOptions

* Update with code review

* Support nested projectId

* Fix test

Commit:
feat: implement the new design updates for the error pages (#21205)

AtofStryker committed:
set remote to primary domain when app is reserved in e2e testing mode

mschile committed:
fixing socket connection

AtofStryker committed:
reset remoteStates on the server when the refreshLifecycle event is run in the data-context package

AtofStryker committed:
Merge branch 'md-10.0-merge' of github.com:cypress-io/cypress into md-10.0-merge

AtofStryker committed:
attempt to increase shouldWithTimeout for consoleProp tests to reduce failures in CI

AtofStryker committed:
Merge branch '10.0-release' of github.com:cypress-io/cypress into md-10.0-merge

AtofStryker committed:
add mount command to expected commands in origin, even though the command is not usable in cy.origin

ryanthemanuel committed:
fix: issue with generating cypress.config.js in projects with type of module (#21323)

Co-authored-by: Zach Bloomquist <git@chary.us>

flotwig committed:
fix: minor tweaks to shared header bar (#21325)

Co-authored-by: Jess <jess@jessicasachs.io>

tgriesser committed:
fix: clear remote data by marking all query data as stale on logout (#21337)

* Add failing test for UNIFY-1652

* Fix for UNIFY-1652

* circle test

flotwig committed:
fix(app): make projectId pull from resolved config (#21324)

AtofStryker committed:
Re-enable page events when connecting to a new spec via an existing
chrome instance within a new tab

try updating stale frame tree

add additional debugging around frame tree updating

simply just re enabled page events for CDP

estrada9166 committed:
feat: hide matches from spec creation (#21293)

* feat: hide matches from spec creation

* Update with designs

* Fix TS

* Update prop name

AtofStryker committed:
fix assertion to update never calls cy.origin test to reference cypress.config.ts over cypress.json in error message

estrada9166 committed:
fix: add min-width and min-height to _x16 icons (#21341)

* fix: add min-width and min-height to _x16 icons

* Apply style for different sizes

* Update icon size

AtofStryker committed:
update cross origin navigation assertions to account for cypress.config.ts path

AtofStryker committed:
improve e2e origin cypress.config.ts error specificity on path

AtofStryker committed:
update spec bridge system test now that chrome/web socket issues are fixed

mschile committed:
increasing the cross:origin:release:html to 1000 to fix "final auth redirects back to approval page - flat" in electron

ZachJW34 committed:
test: add coverage for launchpad header breadcrumbs (#21297)

AtofStryker committed:
allow for a single pixel variance when checking options.animationDistanceThreshold for top and left center positioning for some versions of firefox

AtofStryker committed:
Merge branch 'md-10.0-merge' of github.com:cypress-io/cypress into md-10.0-merge

AtofStryker committed:
Merge branch '10.0-release' of github.com:cypress-io/cypress into md-10.0-merge

ZachJW34 committed:
fix: align 10.x selector playground tooltips with 9.x (#21334)

AtofStryker committed:
bake in additional veriance for top, left, x, y in test for some versions of firefox

AtofStryker committed:
remove links in locales for cy origin and cy session ad the text is now linked to the appropriate command

MuazOthman committed:
fix: Git status ages don't update in real time

MuazOthman committed:
Merge branch '10.0-release' into muaz/UNIFY-1668-bug-real-time-git-updates

tgriesser committed:
fix: cleanup check-ts type failures post merge

AtofStryker committed:
update config package tests to reflect appropriate path

MuazOthman committed:
feat: Spec List: Move git status details to a dynamic file icon [CLOUD-576]

MuazOthman committed:
Merge branch 'muaz/UNIFY-1668-bug-real-time-git-updates' into muaz/CLOUD-576-spec-list-move-git-status

AtofStryker committed:
Merge branch '10.0-release' into md-10.0-merge

AtofStryker committed:
Merge branch 'md-10.0-merge' of github.com:cypress-io/cypress into md-10.0-merge

MuazOthman committed:
Updated commit icon in the spec list

estrada9166 committed:
feat: restart browser/server on config change (#21327)

* feat: restart browser/server on config change

* Add tests

* Refresh server if devServer is changed

* Do not launch browser on start, fix open browser

* Update devServer config pairing w/ Zach

* Update with feedback

flotwig committed:
chore: remove unused parts of design-system, move SCSS to runner-ct (#21345)

AtofStryker committed:
Merge branch '10.0-release' of github.com:cypress-io/cypress into md-10.0-merge

MuazOthman committed:
batched git commands for better performance

tgriesser committed:
chore: remove unused organizationControl graphql field (#21368)

AtofStryker committed:
remove learn run build-prod design-system (bad merge from previous commit)

MuazOthman committed:
Reverted to and fixed shell command for git log

MuazOthman committed:
Merge branch 'muaz/UNIFY-1668-bug-real-time-git-updates' into muaz/CLOUD-576-spec-list-move-git-status

tgriesser committed:
chore: cleanup dead code (#21340)

* chore: cleanup dead code

* fix

* fix

* fix tests

MuazOthman committed:
Added e2e test for real-time git status updates

MuazOthman committed:
Updates to reflect new tooltip design for git info

MuazOthman committed:
Merge branch 'muaz/UNIFY-1668-bug-real-time-git-updates' into muaz/CLOUD-576-spec-list-move-git-status

MuazOthman committed:
Fixed a test stub for git info

AtofStryker committed:
rename 17805_spec.js to 17805.cy.js

MuazOthman committed:
Updated test to cover git info for another cwd

AtofStryker committed:
Merge branch '10.0-release' of github.com:cypress-io/cypress into md-10.0-merge

AtofStryker committed:
update yarn.lock file

tbiethman committed:
chore: Following up on mochaEvent test performance on Windows (#21343)

* Getting clean circle build before changes

* Running retries tests without timeout to see current results

* Moving timeout for retries.mochaEvents. Removing runner.mochaEvents timeout to see what fails.

* Still seeing failures, bumping the resource class to see how that may impact slowness

* D'oh, no medium+ for windows

* Adding back timeouts. Bumping down build resource to medium. Updating mochaEvent tests to navigate directly to spec.

* Updating runSpec

* Updating comments

lmiller1990 committed:
feat: gracefully fail an error when mismatching Cypress version used in migration (#21255)

* error when cypress is not found, or version is mismatching for migration

* add tests

* add defensive try/catch

* Update packages/errors/src/errors.ts

Co-authored-by: Zach Bloomquist <git@chary.us>

* fix error snapshots

* remove usage of resolve-from

* make error more clear around open vs migration

Co-authored-by: Zach Bloomquist <git@chary.us>

lmiller1990 committed:
fix: better language detection in migration/setup workflow (#21220)

* add function to detect lang

* update test files

* update pkg.json

* remove concept of lang from wizard

* update tests

* fix tests and add debugging

* update test

* fix test

* add ts to ts project deps

* update assertion

* empty commit to trigger CI

* fix ts

* add debug log

AtofStryker committed:
get reference to btn inside the trigger for cy.ensureElementIsNotAnimating tests in an attempt to keep strict comparisons in the test

AtofStryker committed:
Merge branch '10.0-release' of github.com:cypress-io/cypress into md-10.0-merge

MuazOthman committed:
fix to address git unit tests

ryanthemanuel committed:
fix: ensure that proxy logs are updated after the xhr has actually completed (#21373)

Co-authored-by: Emily Rohrbough <emilyrohrbough@users.noreply.github.com>

MuazOthman committed:
Merge branch 'muaz/UNIFY-1668-bug-real-time-git-updates' into muaz/CLOUD-576-spec-list-move-git-status

marktnoonan committed:
fix: remove redundant expression from `containerWidth` computed value (#21393)

* avoid depending on open/closed state

* trigger ci

estrada9166 committed:
fix: add custom specPattern if user opt out renaming spec ext (#21344)

* fix: add custom specPattern if user opt out renaming spec ext

* Update test, remove comment

* Fix tests

* Update returned config after migrating folder

* Update with feedback

AtofStryker committed:
Merge branch '10.0-release' into md-10.0-merge

MuazOthman committed:
Replaced using classes in tests to data attr

estrada9166 committed:
fix: CT app icon to have correct sizes (#21356)

* fix: CT app icon to have correct sizes

* Fix TS

Co-authored-by: Mark Noonan <mark@cypress.io>

MuazOthman committed:
Merge branch '10.0-release' into muaz/UNIFY-1668-bug-real-time-git-updates

AtofStryker committed:
Merge branch '10.0-release' into md-10.0-merge

AtofStryker committed:
increase shouldWithTimeout to 3 seconds until #21375 is addressed

AtofStryker committed:
Merge branch 'md-10.0-merge' of github.com:cypress-io/cypress into md-10.0-merge

MuazOthman committed:
Merge pull request #21358 from cypress-io/muaz/UNIFY-1668-bug-real-time-git-updates

Commit:
fix: finalize viewport dropdown in spec runner header (#21352)

AtofStryker committed:
Merge branch '10.0-release' into md-10.0-merge

MuazOthman committed:
Merge branch 'muaz/UNIFY-1668-bug-real-time-git-updates' into muaz/CLOUD-576-spec-list-move-git-status

MuazOthman committed:
Merge branch '10.0-release' into muaz/CLOUD-576-spec-list-move-git-status

MuazOthman committed:
Updated header wording

AtofStryker committed:
Merge pull request #21237 from cypress-io/md-10.0-merge

chore: develop (including cy.origin) merge into 10.0

AtofStryker committed:
Merge branch 'develop' into merge-develop-into-10.0

estrada9166 committed:
fix: do not launch browser if testing type is not configured while switching testing type (#21371)

* fix: do not launch browser if testing type is not configured while switching testing type

* Update test

* Fix test

davidmunechika committed:
test(unify): App - Specs List - Filter Typing Tests (#21315)

* add tests

* stash

* fix escape test

* implement focus on clear search

* update tests

* fix types

* use forwardRef pattern to focus input

* null check for optional ref

* fix tests

Co-authored-by: Lachlan Miller <lachlan.miller.1990@outlook.com>

MuazOthman committed:
Fixed tests and addressed PR comments #21139

ryanthemanuel committed:
chore: fix flaky windows tests (#21354)

tgriesser committed:
fix: remove local warning for the fs.existsSync method (#21426)

AtofStryker committed:
Merge branch '10.0-release' into merge-develop-into-10.0

MuazOthman committed:
Addressed PR comments #21139

MuazOthman committed:
Merge branch '10.0-release' into muaz/CLOUD-576-spec-list-move-git-status

emilyrohrbough committed:
Merge branch 'develop' into merge-develop-into-10.0

# Conflicts:
#	packages/driver/src/cy/commands/sessions/index.ts
#	packages/runner/cypress/integration/sessions.ui.spec.js

MuazOthman committed:
Merge pull request #21139 from cypress-io/muaz/CLOUD-576-spec-list-move-git-status

AtofStryker committed:
Merge branch '10.0-release' into merge-develop-into-10.0

lmiller1990 committed:
chore: change to yarn registry

lmiller1990 committed:
Merge branch 'merge-develop-into-10.0' of github.com:cypress-io/cypress into merge-develop-into-10.0

tgriesser committed:
fix: UNIFY-1625 Runs tab not updating in real time (#21370)

* feat: improved CloudDataSource caching & tests

* feat: pushFragment subscription, for updating the client with remote data (#21408)

* refactor: add pushFragment utility for pushing remote data into the client

* fix: UNIFY-1625 Runs tab not updating in real time (#21412)

* allow returning a Response object in the remoteGraphQLIntercept, handle 401

AtofStryker committed:
Merge branch '10.0-release' into merge-develop-into-10.0

AtofStryker committed:
Merge pull request #21416 from cypress-io/merge-develop-into-10.0

chore: merge develop into 10.0

ZachJW34 committed:
chore: update stubbed cloud types (#21451)

marktnoonan committed:
feat: distinguish app vs launchpad utm_source when using utm params (#21424)

tbiethman committed:
Merge branch 'develop' of https://github.com/cypress-io/cypress into tbiethman/develop-into-10-with-electron

# Conflicts:
#	circle.yml
#	packages/driver/cypress/integration/commands/actions/trigger_spec.js
#	packages/driver/cypress/integration/commands/actions/type_spec.js
#	packages/driver/cypress/integration/commands/files_spec.js
#	packages/electron/package.json
#	packages/launcher/lib/darwin/util.ts
#	packages/launcher/package.json
#	packages/server/__snapshots__/args_spec.js
#	packages/server/__snapshots__/cypress_spec.js
#	system-tests/projects/non-proxied/cypress/fixtures/cy-non-proxied.png
#	yarn.lock

marktnoonan committed:
fix: test-recording instructions in Component Test mode (#21422)

tbiethman committed:
Fixing type, was referencing our internal Cy EventEmitter type before

estrada9166 committed:
fix: remove breaking options from testing type on migration (#21437)

* Add failing test

* fix test

* Refactor code

* Update test

tbiethman committed:
More types

tbiethman committed:
MORE TYPES

tbiethman committed:
Updating system test image

tbiethman committed:
Merge branch '10.0-release' of https://github.com/cypress-io/cypress into tbiethman/develop-into-10-with-electron

estrada9166 committed:
fix: use correct path for scaffolding spec on CT (#21411)

* fix: use correct path for scaffolding spec on CT

* Refactor code

* Update with feedback

ZachJW34 committed:
chore: remove unused codeGenGlobs (#21438)

ZachJW34 committed:
test: fix flaky cy-in-cy selector validity test (#21360)

tbiethman committed:
Updating origin viewport tests to account for some Firefox differences

tbiethman committed:
Using once to listen to failures and finish tests to combat flake with done executing multiple times

tbiethman committed:
Adding mac/windows builds

estrada9166 committed:
fix: migrate multiples projects when in global mode (#21458)

* fix: migrate multiples projects when in global mode

* Add test

* Update test

tbiethman committed:
Testing if this work around is causing new failures in CI

MuazOthman committed:
fix: Files with special characters aren't handled properly in GitDataSource on Windows (#21436)

marktnoonan committed:
test: remove flaky click from navigation spec (#21466)

ZachJW34 committed:
fix: ct testing support for node 17+ (#21430)

tbiethman committed:
Merge branch '10.0-release' of https://github.com/cypress-io/cypress into tbiethman/develop-into-10-with-electron

# Conflicts:
#	circle.yml

tbiethman committed:
Revert "Testing if this work around is causing new failures in CI"

This reverts commit 1abd0efa85237c858db517c98436ba651f036ca4.

MuazOthman committed:
fix: Change the capitalization of Component Specs (#21460)

Commit:
chore: replace use of yellow by orange (#21429)

Co-authored-by: Blue F <blue@cypress.io>

flotwig committed:
test(driver): skip failing proxy-logging test in Electron (#21498)

tbiethman committed:
Merge branch '10.0-release' of https://github.com/cypress-io/cypress into tbiethman/develop-into-10-with-electron

estrada9166 committed:
fix: handle errors on plugins migration child process (#21442)

* fix: handle errors on plugins migration child process

* Update error handler

* Remove console.log

estrada9166 committed:
fix: terminal error message for non migrated config (#21467)

* fix: terminal error message for non migrated config

* fix: terminal error message for non migrated config

estrada9166 committed:
feat: display full browser info (#21452)

* feat: display full browser info

* Update test

* Fix TS

* Remove browser path from test

lmiller1990 committed:
feat: automatically find/merge config for component testing (#21432)

* wip: add comment regarding webpack config

* wip: various

* wip

* fix tests

* lint

* fix tests

* add tests

* add tests

* fix tests

* update imports

* make test more reliable

* parse correct flag

* minor refactor

* replace fs-extra with fs

Co-authored-by: Zachary Williams <zachjw34@gmail.com>

tbiethman committed:
Merge branch '10.0-release' into tbiethman/develop-into-10-with-electron

tbiethman committed:
Revert "Using once to listen to failures and finish tests to combat flake with done executing multiple times"

This reverts commit ce58eae17da6b552c1165b17dec9780b33d38dfc.

Commit:
fix: Unify 1232 the keyboard shortcuts button is not keyboard accessible (#21409)

emilyrohrbough committed:
fix(driver): remove extra sessions log (#21457)

flotwig committed:
chore(proxy): improve debug log experience (#21489)

tbiethman committed:
More targeted zip install

tbiethman committed:
Merge branch '10.0-release' of https://github.com/cypress-io/cypress into tbiethman/develop-into-10-with-electron

estrada9166 committed:
fix: display cy.mount command log (#21500)

* fix: display cy.mount log

* Update with feedback

tbiethman committed:
Better targeting origin workarounds to combat flake

tbiethman committed:
Merge branch '10.0-release' of https://github.com/cypress-io/cypress into tbiethman/develop-into-10-with-electron

lmiller1990 committed:
fix: do not include non specs when globbing using --spec in run mode (#21462)

* fix: do not include non specs when globbing using --spec in run mode

* update pattern

* fix test

* update cdp spec

* wip - updates

* update snapshots

* update non-proxied spec

* update snaps

* update snaps

* rename specs

* update spec

* wip - snapshots

* snaps

* snaps

* update spec names

* update

* update test

* snaps

* update snap and spec

* snaps

* correct spec pattern

* snaps

* revert

* update spec and snapshots

* fix test

* update tests

* fix test

* update test

* update snapshot

* update snaps

* include coffee in specPattern

* update snapshots

* update snaps

* rename specs

* snaps

* update test

* update snapshot

* update

* snaps

* update snap

* update snaps

* fix test

* unskip test

* snaps

* add test

lmiller1990 committed:
fix: do not clean test fixtures (#21528)

tbiethman committed:
Merge branch '10.0-release' into tbiethman/develop-into-10-with-electron

tbiethman committed:
Merge pull request #21456 from cypress-io/tbiethman/develop-into-10-with-electron

chore: merge develop into 10.0-release (w/ electron/node updates)

flotwig committed:
chore: remove unused find_system_node util (#21529)

* chore: remove unused find_system_node util

* bump

AtofStryker committed:
fix: electron slowness within cy origin (#21445)

* fix iframe performance issues in electron

* reduce 1000ms to 300ms for cross:origin:release:html

* reduce shouldWithTimeout from 3000ms to 250ms

* Update system-tests/projects/e2e/cypress/e2e/spec_bridge.cy.ts

Co-authored-by: Chris Breiding <chrisbreiding@users.noreply.github.com>

* add some regression tests around screenshot size when taking screenshots within cy.origin

* move screenshot e2e tests from system-tests into the driver

* move spec bridge system test into driver test

* add additional comments to spec bridge iframe scss for caution

* resolve snapshots directory correctly in open vs run mode

* revert this commit to see if assets are being tampered with or just missing in CI

* fix check:screenshot:size to always point to correct screenshot path

* remove isCypressRunMode from cypress config as internal flag as we no longer need it

* add check:screenshot:size to task.cy.js registered plugin assertion

Co-authored-by: Chris Breiding <chrisbreiding@users.noreply.github.com>

estrada9166 committed:
fix: update dropdown shadow (#21519)

Co-authored-by: Mark Noonan <mark@cypress.io>

ZachJW34 committed:
feat: add runs duration, relative created at and style tweaks (#21410)

Co-authored-by: Tim Griesser <tgriesser10@gmail.com>
Co-authored-by: Mark Noonan <mark@cypress.io>

ZachJW34 committed:
fix: use resolved port for vite (#21490)

davidmunechika committed:
fix(unify): Have better default filename than `filename` for create spec file windows (#21396)

* fix: change default filename for create spec

* add test

* fix typo

* fix unit tests

* fix unit tests

* fix types

* fix runner test

* refactor defaultSpecFileName

* add e2e tests

* refactor replaceWildCard

Co-authored-by: Tyler Biethman <tbiethman@users.noreply.github.com>

* update spec pattern

* Removing extra codeGenBlobs missed in merge

* Updating tests to reflect default component name change

* Let's get a windows build too

* Windows test fixes

* Addressing Zach's feedback. Fixing type.

* Update packages/app/cypress/e2e/specs_list_component.cy.ts

Co-authored-by: Rachel <rachel@cypress.io>

* Using mountFragment for new component test

Co-authored-by: Tyler Biethman <tbiethman@users.noreply.github.com>
Co-authored-by: Tyler Biethman <tbiethman@gmail.com>
Co-authored-by: Rachel <rachel@cypress.io>

emilyrohrbough committed:
chore: setup node events / server when e2e test type is selected (#21537)

* chore: setup node events / server when e2e test type is selected

* Update packages/driver/cypress.config.ts

Co-authored-by: Tim Griesser <tgriesser10@gmail.com>

Co-authored-by: Tim Griesser <tgriesser10@gmail.com>

lmiller1990 committed:
chore: document next.js specific component template (#21507)

* chore: document next.js specific component template

* update test

* remove obsolete test

* update test

* correct path for glob

* fix test

ZachJW34 committed:
fix: next 12.1.6 support and add tests (#21515)

ryanthemanuel committed:
chore: fix windows tests in 10.0 (#21427)

Co-authored-by: Lachlan Miller <lachlan.miller.1990@outlook.com>
Co-authored-by: Tyler Biethman <tbiethman@users.noreply.github.com>
Co-authored-by: Tyler Biethman <tbiethman@gmail.com>

Commit:
fix: Spec renaming opt out improvements (#21245)

* Don't display spec renaming when user has opted out

* Combine OptOutModal steps

* First round of review feedback

* Fix for migration test

* Review comments

* Fix merge additional errors

* More review feedback

* Try to fix broken unit test

* Fix additional e2e test

* Restore missing unit test files

flotwig committed:
feat(app): use longest common prefix for default spec filename (#21527)

Co-authored-by: Tim Griesser <tgriesser10@gmail.com>

flotwig committed:
feat: launch project if `cypress open --e2e/--ct --browser` is used (#21538)

Co-authored-by: Lachlan Miller <lachlan.miller.1990@outlook.com>

tbiethman committed:
chore: Removing workaround for long Cypress verification times (#21541)

* chore: Removing workaround for long Cypress verification times

* Blank commit for Percy build

* Getting a build

* Need the mac build

* Missed a few!

* We apparently still need this one.

marktnoonan committed:
feat: migration landing page (#21349)

flotwig committed:
Merge remote-tracking branch 'origin/develop' into 10.0-release

tgriesser committed:
fix: UNIFY-1611, add next to framework options & add tests (#21557)

* fix: UNIFY-1611, add next to framework options & add tests

tgriesser committed:
fix: revert "feat: automatically find/merge config for component testing (#21432)" (#21560)

This reverts commit a355ad61593df38c75e9df109b2cce740c466a61.

tgriesser committed:
refactor: reduce flake in launchpad tests, return null for videoEmbedHtml if not needed (#21565)

* refactor: reduce flake in migration.cy.ts, return null for videoEmbedHtml if migration is not needed

* Fix UNIFY-1740: guard against double submit of mutation

* don't hit the videoHtmlEmbed link more than once

* move withCtx after scaffold

* Stub the on link response, add test for caching behavior

tbiethman committed:
fix: Ensuring spec file presence prior to webpack-dev-server compilation (#21550)

* fix: Ensuring spec file presence prior to webpack-dev-server compilation

* Clean up test a bit

* Async beforeCompile

* Adding unit test for new behavior

* Moving e2e test to react specs for better project coverage

ZachJW34 committed:
fix: disable ct testing when run via binary (#21540)

lmiller1990 committed:
chore: sort the frameworks select dropdown in a more logical way (#21553)

AtofStryker committed:
fix: do not allow experimentalSessionAndOrigin to be available in CT … (#21588)

* fix: do not allow experimentalSessionAndOrigin to be available in CT per https://github.com/cypress-io/cypress/issues/21573

* remove experimentalSessionAndOrigin config as it is read only

* move runner session tests under own system test project to turn on experimentalSessionAndOrigin flag

* refactor session ui runner tests to point to runner-session-e2e-specs folder

* fix cypress_spec.js to account for experimentalSessionAndOrigin options

* remove experimentalSessionAndOrigin from env args

tbiethman committed:
fix: Successfully create new specs files that do not have a known extension (#21593)

* fix: Successfully create new specs files that do not have known a known extension

* Adding e2e tests.

* Updating verbiage

* PR feedback

ZachJW34 committed:
Merge remote-tracking branch 'origin/develop' into zachw/merge-develop-10.0

estrada9166 committed:
fix: handle git watch error (#21600)

* fix: handle git watch error

* Add test

* Cleanup sinon

ZachJW34 committed:
Merge branch '10.0-release' into zachw/merge-develop-10.0

ZachJW34 committed:
Merge remote-tracking branch 'origin/develop' into zachw/merge-develop-10.0

ZachJW34 committed:
Merge pull request #21609 from cypress-io/zachw/merge-develop-10.0

flotwig reviewed (approved):
How can I approve more than once? ✅✅

flotwig committed:
test(launchpad): skip failure due to recaptcha on windows, enable more windows jobs (#21620)

Co-authored-by: Tyler Biethman <tbiethman@users.noreply.github.com>
